docs/proposals/10-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,146 @@
+
+Since each operating system distribution has different methods of installing software (apt on ubuntu, zypper on SuSE, Torcx on CoreOS/Flatcar), we will enhance the operating system extensions to add a standard `installSoftware` script which can abstract the mechanics of installing new software on a node. The new runtime extensions will use this new functionality to install the relevant runtimes to the worker nodes.
+
+## Design Details
+
+1. Change the nodes container runtime to work with CRI and containerd by default.
+    1. In order to configure each worker machine in the cluster to work with CRI, the following configurations should be done:
+        1. Add kubelet execution flags:
+            1. --container-runtime=remote
+            2. --container-runtime-endpoint=unix:///.../containerd.socket
+        2. Make sure that containerd configuration file exist in path /etc/containerd/config.toml and it is configured to enable the CRI plugin (In the containerd config.toml "cri" plugin must not be disabled).
+        
+    2. Containerd socket path and containerd configurations are different for each OS. To make sure the configurations above are set in each worker machine, each OS extension would be responsible to configure them during the reconciliation of the 
+       OperatingSystemConfig when it creates the cloud init.
+        1. os-ubuntu - 
+            1. Add the following flags to the kubelet execution command:
+                1. --container-runtime=remote
+                2. --container-runtime-endpoint=unix:///run/containerd/containerd.sock
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the current default configuration (containerd config default).
+        2. os-coreos - 
+            1. Add the following flags to the kubelet execution command:
+               1. --container-runtime=remote
+               2. --container-runtime-endpoint=unix:///var/run/docker/libcontainerd/docker-containerd.socket
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the default configuration.
+            3. Override the default containerd.service unit. Create a new unit in: /etc/systemd/system/containerd.service unit that will start the containerd service with a configuration file path: /etc/containerd/config.toml.
+        3. os-suse-jeos - 
+           1. Add the following flags to the kubelet execution command:
+              1. --container-runtime=remote
+              2. --container-runtime-endpoint=unix:///var/run/docker/containerd/docker-containerd.sock 
+           2. Create a containerd configuration file: /etc/containerd/config.toml based on the default configuration.
+        
+    3. Docker-monitor daemon and conatinerd service. For CRI it is possible to send parameters to Kubelet to do the log rotation:
+       CRIContainerLogRotation=true
```

Comment:
Can you explain this point? How will you adapt the Docker-Monitor? Do we still need it at all?
This `CRIContainerLogRotation` is a feature gate? So, the OS controller will set it always, correct? I think this section needs a little bit more descriptive text.

Reply:
I added